extension/content.css

```diff
@@ -399,21 +403,50 @@ svg.octicon.octicon-star.dashboard-event-icon {
 }
 
 .reaction-summary-item div,
+.reaction-summary-item div a,
 .reaction-summary-item div img {
 	display: inline-block;
 }
 
+.reaction-summary-item div a {
+	transition: margin-left 0.2s;
+}
+
 .reaction-summary-item div img,
 .reaction-summary-item div span {
 	width: 20px;
 	height: 20px;
-	margin-left: 3px;
 	vertical-align: middle;
 }
+
 .reaction-summary-item div img {
+	color: #FFF; /* affects box-shadow */
+	box-shadow: 0 0 0 2px;
 	border-radius: 3px;
 }
 
+.reaction-summary-item.user-has-reacted div img {
+	color: #f2f8fa;
+}
+
+/* overlap reaction avatars when there are 5+ types of reactions */
+.reaction-summary-item:first-child:nth-last-child(n+5) a:nth-child(n+2),
```

Comment:
If we're putting the `margin-left` on the `.reaction-summary-item div a` as I mentioned above then we can use a `:not` here and remove the block for "un-overlap on hover":

```css
/* overlap reaction avatars when there are 5+ types of reactions */
.reaction-summary-item:first-child:nth-last-child(n+5):not(:hover) a:nth-child(n+2),
.reaction-summary-item:first-child:nth-last-child(n+5) ~ *:not(:hover) a:nth-child(n+2) {
	margin-left: -12px;
}
```